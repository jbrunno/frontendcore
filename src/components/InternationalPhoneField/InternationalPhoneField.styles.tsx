import { styled } from '@mui/material/styles';
import PhoneInput from 'react-phone-input-2';
import Typography from '../Typography/Typography';

export const TypographyError = styled(Typography)`
  margin-left: ${({ theme }) => theme.fn.spacing(14)};
`;

export const PhoneInputStyled = styled(PhoneInput)<{
  color: 'primary' | 'secondary' | 'error' | 'info' | 'success' | 'warning';
  isFocused?: boolean;
  size?: 'small' | 'medium';
}>(({ theme, isValid, color, isFocused, size = 'medium' }) => {
  let colorText = theme.palette.text.secondary;
  let padding = theme.fn.spacing(18.5, 14, 18.5, 58);

  if (!isValid) {
    colorText = theme.palette.error.main;
  } else if (isFocused) {
    colorText = theme.palette[color].main;
  }

  if (size === 'small') {
    padding = theme.fn.spacing(8.5, 14, 8.5, 58);
  }

  return {
    '&.react-tel-input .special-label': {
      top: theme.fn.spacing(-9),
      left: theme.fn.spacing(7),
      color: colorText,
      fontSize: theme.typography.caption.fontSize,
      fontWeight: theme.typography.caption.fontWeight,
      lineHeight: theme.typography.caption.lineHeight,
      letterSpacing: theme.typography.caption.letterSpacing,
    },
    '&.react-tel-input .form-control': {
      width: '100%',
      borderRadius: theme.shape.borderRadius,
      borderColor: colorText,
      height: theme.fn.size(40),
      boxShadow: isFocused
        ? `0 0 0 ${theme.fn.spacing(1)} ${colorText}`
        : 'none',
      padding,
    },
    '&.react-tel-input .form-control:focus': {
      borderColor: colorText,
      boxShadow: `0 0 0 ${theme.fn.spacing(1)} ${colorText}`,
    },
  };
});
