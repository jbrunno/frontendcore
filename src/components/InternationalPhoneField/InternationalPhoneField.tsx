import { PhoneInputProps } from 'react-phone-input-2';
import pt from 'react-phone-input-2/lang/pt.json';
import 'react-phone-input-2/lib/material.css';
import Box from '../Box/Box';
import { useInternationalPhoneField } from './hooks/useInternationalPhoneField';
import {
  PhoneInputStyled,
  TypographyError,
} from './InternationalPhoneField.styles';

export type InternationalPhoneFieldProps = Omit<PhoneInputProps, 'isValid'> & {
  color?: 'primary' | 'secondary' | 'error' | 'info' | 'success' | 'warning';
  size?: 'small' | 'medium';
  focused?: boolean;
  error?: string;
  className?: string;
};

export function InternationalPhoneField({
  onFocus,
  onBlur,
  focused,
  color = 'primary',
  error,
  size = 'medium',
  className,
  ...props
}: InternationalPhoneFieldProps) {
  const { isFocused, handleFocus, handleBlur } = useInternationalPhoneField({
    onFocus,
    onBlur,
    focused,
  });

  return (
    <Box className={className}>
      <PhoneInputStyled
        {...props}
        onFocus={handleFocus}
        onBlur={handleBlur}
        color={color}
        localization={pt}
        isFocused={isFocused}
        isValid={!error}
        size={size}
      />
      {error && (
        <TypographyError variant="helperText" color="error">
          {error}
        </TypographyError>
      )}
    </Box>
  );
}

export default InternationalPhoneField;
