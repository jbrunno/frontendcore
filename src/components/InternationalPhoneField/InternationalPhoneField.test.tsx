import React from 'react';
import { render } from '@test-utils/component-testing';
import { InternationalPhoneField } from './InternationalPhoneField';

test('renders the InternationalPhoneField', () => {
  const { container } = render(<InternationalPhoneField />);
  expect(container).toMatchSnapshot();
});
