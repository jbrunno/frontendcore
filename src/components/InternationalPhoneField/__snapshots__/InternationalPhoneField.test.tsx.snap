// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the InternationalPhoneField 1`] = `
.emotion-1.react-tel-input .special-label {
  top: -0.5625rem;
  left: 0.4375rem;
  color: #3B3B4199;
  font-size: 0.75rem;
  font-weight: 400;
  line-height: 1.125rem;
  letter-spacing: 0.0106rem;
}

.emotion-1.react-tel-input .form-control {
  width: 100%;
  border-radius: 4px;
  border-color: #3B3B4199;
  height: calc(1.75rem + 12px);
  box-shadow: none;
  padding: 1.15625rem 0.875rem 1.15625rem 3.625rem;
}

.emotion-1.react-tel-input .form-control:focus {
  border-color: #3B3B4199;
  box-shadow: 0 0 0 0.0625rem #3B3B4199;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class=" react-tel-input  emotion-1 emotion-2"
    >
      <div
        class="special-label"
      >
        Phone
      </div>
      <input
        class="form-control "
        placeholder="1 (702) 123-4567"
        type="tel"
        value=""
      />
      <div
        class="flag-dropdown "
      >
        <div
          aria-haspopup="listbox"
          class="selected-flag"
          role="button"
          tabindex="0"
          title=""
        >
          <div
            class="flag 0"
          >
            <div
              class="arrow"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
